sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpMessageConverter.cs

```diff
@@ -218,62 +207,72 @@ public static AmqpMessage SBMessageToAmqpMessage(SBMessage sbMessage)
         public static ServiceBusReceivedMessage AmqpMessageToSBMessage(AmqpMessage amqpMessage, bool isPeeked = false)
         {
             Argument.AssertNotNull(amqpMessage, nameof(amqpMessage));
+            AmqpAnnotatedMessage annotatedMessage;
 
-            ServiceBusReceivedMessage sbMessage = amqpMessage.ToServiceBusReceivedMessage();
-            var sections = amqpMessage.Sections;
+            if ((amqpMessage.BodyType & SectionFlag.Data) != 0 && amqpMessage.DataBody != null)
```

Comment:
nit:  I think that you could use the [`HasFlag`](https://docs.microsoft.com/en-us/dotnet/api/system.enum.hasflag?view=netcore-3.1) method here, which may improve readability.

Reply:
My understanding is there is still a performance overhead to using HasFlag.